Analyze sequences

tswayd opened the pull request:
Hi professor, Michali, and Kory. Thank you for this challenging assignment. If you could review my code and output that would be very much appreciated. Thank you. @naupaka @mhaliz @melt6457

tswayd committed:
Initial commit of sequence process script

tswayd committed:
Delete unecessary script

tswayd committed:
Commit to new branch

tswayd committed:
Update .profile and process_sequences.sh

tswayd committed:
Add code for first two problems

tswayd committed:
Save this thing I don't know

tswayd committed:
Add new code to find sequence number

tswayd committed:
Add code for optional challenge

tswayd committed:
Update code and include websites

tswayd committed:
Add defensive argument spellcheck

tswayd committed:
Redirect output into file

tswayd committed:
Delete old script

tswayd committed:
Change redirect into a pipe redirect

tswayd committed:
Test bash script

tswayd committed:
Commit for unknown reason

tswayd committed:
Redirect to output dir

tswayd committed:
Delete file I don't know why

tswayd committed:
Redirect echo statements to file

tswayd committed:
Initial commit of output file

tswayd committed:
Change redirect from tee -a to >>

tswayd committed:
Commit of secondary output file

tswayd committed:
Commit for unknown reason

tswayd committed:
Change redirect back from >> to | tee -a

tswayd committed:
Add line breaks and descriptor

tswayd committed:
Commit tertiary test of script

tswayd committed:
"Add and remove line breaks and words"

tswayd committed:
Commit of output

tswayd committed:
Final commit of output

naupaka commented:
hi @tswayd -- I am assigning each review to one of us, and allocating them based on the order they come in, so that we know who is responsible for what. I've cleared your assignee designations and will add the reviewer soon.

tswayd commented:
Ok thank you I was unsure if I needed to tag or assign so I just did both my apologies.

naupaka commented:
No worries. We're just trying to keep track of a lot of moving parts and this is the system we're using to handle things.

naupaka commented on code/process_sequences.sh:
```diff
@@ -0,0 +1,41 @@
+# dttsway@dons.usfca.edu
+# 2020-10-18
+
+# resources I consulted to build this code:
```
Great! I often do this too. Keeps the tips right next to the code

naupaka commented on code/process_sequences.sh:
```diff
@@ -0,0 +1,41 @@
+# https://www.cyberciti.biz/faq/run-execute-sh-shell-script/
+# https://askubuntu.com/questions/420981/how-do-i-save-terminal-output-to-a-file
+
+FILE="/blast-db/sars-cov-2-seq-data/2020-10-15_ncbi_sars_cov_2_sequences.fasta.gz"
```
Since the goal is to have the script take in the parameter from the command line, you should remove this, since it means the script will never work with any other file, since you set the file name inside the script.

naupaka commented on code/process_sequences.sh:
```diff
@@ -0,0 +1,41 @@
+
+if [ $# -gt 1 ]
+then
+echo "There is more than the one require argument provided"
```
Indent this echo statement just like you did for the first one, for readibility.

naupaka commented on code/process_sequences.sh:
```diff
@@ -0,0 +1,41 @@
+exit 1
+fi
+# echo statement to describe contents:
+echo "This is some information about the sequences from the Sars-Cov-2 database" > output/SeqSort.txt
```
Don't send these to a file, just print them to the screen, so that when you redirect the output of this script, it has the description in the same file and if you don't redirect the output, then the helpful information is printed to the screen.

naupaka commented on code/process_sequences.sh:
```diff
@@ -0,0 +1,41 @@
+echo "This is some information about the sequences from the Sars-Cov-2 database" > output/SeqSort.txt
+
+# code to count total number of sequences in file
+echo "Number of sequences in file:" >> output/SeqSort.txt
```
Same as above. Remove the redirect bit at the end.

naupaka commented on code/process_sequences.sh:
```diff
@@ -0,0 +1,41 @@
+
+# code to count total number of sequences in file
+echo "Number of sequences in file:" >> output/SeqSort.txt
+bioawk -cfastx 'END{print NR}' $FILE | tee -a output/SeqSort.txt 
```
Same as earlier. Remove the tee bit since you can just redirect the entire output at once when you run the script.

naupaka commented on code/process_sequences.sh:
```diff
@@ -0,0 +1,41 @@
+
+# code to parse out sequences by country, count by unique value, and then sort in descending order
+echo "Number of sequences by country in descending order:" >> output/SeqSort.txt
+bioawk -c fastx '{print $4}' $FILE | awk '{split($0,a,"|");print a[3]}' | sort | uniq -c | sort -nr | tee -a output/SeqSort.txt
```
Fancy! You can also use `cut` to do something similar here. Also remove the tee part so that this just goes to the screen.

naupaka commented on code/process_sequences.sh:
```diff
@@ -0,0 +1,41 @@
+bioawk -c fastx '{print $4}' $FILE | awk '{split($0,a,"|");print a[3]}' | sort | uniq -c | sort -nr | tee -a output/SeqSort.txt
+
+# code to count number of sequences below and above 29,000 bp 
+bioawk -cfastx 'BEGIN{shorter = 0} {if (length($2) < 29000) shorter += 1} END {print "Number of sequences shorter than 29K bp:", shorter}' $FILE | tee -a output/SeqSort.txt
```
Same as above. Remove the `tee` part.

naupaka reviewed (changes requested):
Nice job!! I make some suggestions for you to fix, nothing that should take more than a couple minutes.

naupaka commented:
Also! Don't forget to run `find . -name "*.sh" | xargs shellcheck` once you made your changes to check that all the scripts are ok.

tswayd commented:
Hi professor thank you for the suggestions and comments! I removed the redirects, added the indent, removed the "FILE=", redefined the "$FILE" as "$1", and passed a shellcheck. I also deleted the old output file and made a newer one. I hope everything is to a satisfactory level, am going to go ahead and close pull request.

naupaka commented:
@tswayd too late now, but proper PR protocol is that you wait for the PR to be approved by the reviewer before merging. It's hard to un-merge something, and there may be other things that need to get fixed after the first round of changes.

naupaka commented:
I also do not see your new commits here, most likely because you did not push them, but it's tricky now because the PR is merged.

naupaka commented:
Please push those new commits to GH so I can look them over.

tswayd commented:
Oh woops very sorry about that. Ok will push.

naupaka commented:
One other thing you should do it update your README log

tswayd commented:
Roger that sorry I forgot a lot of these things I guess I'm still pretty shaky on the ins and outs of GitHub and documenting my work in general. Will work on these skills for the remainder of the course.

naupaka commented:
No worries! There are a ton of different things to keep track of and I'm sure a lot of it is new. The goal is to learn good habits
